piano-hero/mobile/src/main/java/com/novoda/pianohero/GamePresenter.java

```diff
@@ -0,0 +1,44 @@
+package com.novoda.pianohero;
+
+class GamePresenter implements GameMvp.Presenter {
+
+    private final GameMvp.Model gameModel;
+    private final GameMvp.View view;
+
+    GamePresenter(GameMvp.Model gameModel, GameMvp.View view) {
+        this.gameModel = gameModel;
+        this.view = view;
+    }
+
+    @Override
+    public void onCreate() {
+        RoundViewModel viewModel = gameModel.startGame();
+        view.showRound(viewModel);
+    }
+
+    @Override
+    public void onNotesPlayed(Note... notes) {
+        RoundViewModel viewModel = gameModel.onNotesPlayed(new GameMvp.Model.Callback() {
```

Comment:
So the whole business logic happens in the `model`, and it callbacks to the presenter when the game succeeds so that the presenter knows what to do going forward 👍 

What I find it weird is that this method `onNotesPlayed` returns a viewModel synchronously, but at the same time attaches this callback that async says when the game is complete. It seems to me that the attaching of the callback could be done separately ? 

`gameModel.setOnSequenceComplete(new GameMvp.Model.Callback())`